relay-server/src/utils/rate_limits.rs

```diff
@@ -850,4 +936,110 @@ mod tests {
+        let mut envelope = envelope![Transaction];
+        set_extracted(&mut envelope, ItemType::Transaction);
+        let config = config_with_tx_metrics();
+
+        let mut mock = MockLimiter::default().deny(DataCategory::Transaction);
+        let limiter = EnvelopeLimiter::new(Some(&config), |s, q| mock.check(s, q));
+        let (enforcement, limits) = limiter.enforce(&mut envelope, &scoping()).unwrap();
+
+        assert!(limits.is_limited());
+        assert!(enforcement.event_metrics.is_active());
+        assert!(enforcement.event.is_active());
+    }
+
+    #[test]
+    fn test_enforce_transaction_no_indexing_quota() {
+        let mut envelope = envelope![Transaction];
+        let config = config_with_tx_metrics();
+
+        let mut mock = MockLimiter::default().deny(DataCategory::TransactionIndexed);
+        let limiter = EnvelopeLimiter::new(Some(&config), |s, q| mock.check(s, q));
+        let (enforcement, limits) = limiter.enforce(&mut envelope, &scoping()).unwrap();
+
+        // NOTE: Since metrics have not been extracted on this item, we do not check the indexing
+        // quota. Basic processing quota is not denied, so the item must pass rate limiting.
+
+        assert!(!limits.is_limited());
+        assert!(!enforcement.event_metrics.is_active());
+        assert!(!enforcement.event.is_active());
+        mock.assert_call(DataCategory::Transaction, Some(0));
+    }
+
+    #[test]
+    fn test_enforce_transaction_metrics_extracted_no_indexing_quota() {
+        let mut envelope = envelope![Transaction];
+        set_extracted(&mut envelope, ItemType::Transaction);
+        let config = config_with_tx_metrics();
+
+        let mut mock = MockLimiter::default().deny(DataCategory::TransactionIndexed);
+        let limiter = EnvelopeLimiter::new(Some(&config), |s, q| mock.check(s, q));
+        let (enforcement, limits) = limiter.enforce(&mut envelope, &scoping()).unwrap();
+
+        assert!(limits.is_limited());
+        assert!(!enforcement.event_metrics.is_active());
+        assert!(enforcement.event.is_active());
+        mock.assert_call(DataCategory::Transaction, Some(0));
+        mock.assert_call(DataCategory::TransactionIndexed, Some(1));
+    }
+
+    #[test]
+    fn test_enforce_transaction_attachment_enforced() {
+        let mut envelope = envelope![Transaction, Attachment];
+        let config = config_with_tx_metrics();
+
+        let mut mock = MockLimiter::default().deny(DataCategory::Transaction);
+        let limiter = EnvelopeLimiter::new(Some(&config), |s, q| mock.check(s, q));
+
+        let (enforcement, _limits) = limiter.enforce(&mut envelope, &scoping()).unwrap();
+
+        assert!(enforcement.event.is_active());
+        assert!(enforcement.attachments.is_active());
+        mock.assert_call(DataCategory::Transaction, Some(0));
+        mock.assert_call(DataCategory::Attachment, None);
+    }
+
+    #[test]
+    fn test_enforce_transaction_attachment_enforced_metrics_extracted() {
```

Comment:
```suggestion
    fn test_enforce_transaction_attachment_enforced_metrics_extracted_indexing_quota() {
```
not sure? To be consistent with other test names.